iOS/UatsApp/LoginVC.swift

```diff
@@ -0,0 +1,180 @@
+            var urlData: NSData? = NSURLConnection.sendSynchronousRequest(request, returningResponse:&response, error:&reponseError)
+            
+            if ( urlData != nil ) {
+                let res = response as! NSHTTPURLResponse!;
+                
+                NSLog("Response code: %ld", res.statusCode);
+                
+                if (res.statusCode >= 200 && res.statusCode < 300)
+                {
+                    var responseData:NSString  = NSString(data:urlData!, encoding:NSUTF8StringEncoding)!
+                    
+                    NSLog("Response ==> %@", responseData);
+                    
+                    var error: NSError?
+                    
+                    let jsonData:NSDictionary = NSJSONSerialization.JSONObjectWithData(urlData!, options:NSJSONReadingOptions.MutableContainers , error: &error) as! NSDictionary
+                    
+                    
+                    let success:NSInteger = jsonData.valueForKey("success") as! NSInteger
+                    
+                    //[jsonData[@"success"] integerValue];
+                    
+                    NSLog("Success: %ld", success);
+                    
+                    if(success == 1)
+                    {
+                        NSLog("Login SUCCESS");
+                        var alertView:UIAlertView = UIAlertView()
+                        alertView.title = "Success!"
+                        alertView.message = "You are logged in!"
+                        alertView.delegate = self
+                        alertView.addButtonWithTitle("OK")
+                        alertView.show()
+                        var prefs:NSUserDefaults = NSUserDefaults.standardUserDefaults()
+                        prefs.setObject(username, forKey: "USERNAME")
+                        prefs.setInteger(1, forKey: "ISLOGGEDIN")
+                        prefs.synchronize()
+                        self.performSegueWithIdentifier("goApp", sender: self)
+                    
+                      //  self.dismissViewControllerAnimated(true, completion: nil)
+                    } else {
+                        var error_msg:NSString
+                        
+                        if jsonData["error_message"] as? NSString != nil {
+                            error_msg = jsonData["error_message"] as! NSString
+                        } else {
+                            error_msg = "Unknown Error"
+                        }
+                        var alertView:UIAlertView = UIAlertView()
+                        alertView.title = "Sign in Failed!"
+                        alertView.message = error_msg as String
+                        alertView.delegate = self
+                        alertView.addButtonWithTitle("OK")
+                        alertView.show()
+                        
+                    }
+                    
+                } else {
+                    var alertView:UIAlertView = UIAlertView()
```

Comment:
try to extract the UIAlertView code into a global method.. this is what is called code duplication (having the same code in x places)